wundergraph_cli/src/print_schema/mod.rs

```diff
@@ -311,41 +311,51 @@ mod tests {
         std::thread::sleep(std::time::Duration::from_secs(1));
 
         let query = "{\"query\": \"{ Users { id  name  } } \"}";
-        let mut r = client
-            .post(&format!("http://{}/graphql", listen_url))
-            .body(query)
-            .header(
-                reqwest::header::CONTENT_TYPE,
-                reqwest::header::HeaderValue::from_static("application/json"),
-            )
-            .send()
-            .unwrap();
-        insta::assert_json_snapshot!(r.json::<serde_json::Value>().unwrap());
-
         let mutation = r#"{"query":"mutation CreateUser {\n  CreateUser(NewUser: {name: \"Max\"}) {\n    id\n    name\n  }\n}","variables":null,"operationName":"CreateUser"}"#;
-        let mut r = client
-            .post(&format!("http://{}/graphql", listen_url))
-            .body(mutation)
-            .header(
-                reqwest::header::CONTENT_TYPE,
-                reqwest::header::HeaderValue::from_static("application/json"),
-            )
-            .send()
-            .unwrap();
-        insta::assert_json_snapshot!(r.json::<serde_json::Value>().unwrap());
+        let t1 = request_test(&client, &listen_url, query);
+        let t2 = request_test(&client, &listen_url, mutation);
+        let t3 = request_test(&client, &listen_url, query);
 
-        let mut r = client
+        child.kill().unwrap();
+        child.wait().unwrap();
+
+        if !t1 || !t2 || !t3 {
+            panic!("round_trip failed")
+        }
+    }
+
+    fn request_test(client: &reqwest::Client, listen_url: &str, body: &'static str) -> bool {
```

Comment:
I would prefer to just return a result here and bubble up any potential errors. Then we could just do a `t1.is_err()` instead of `!t1` in the test above. I think this would make the code much clearer.

Reply:
Indeed!